R/fit.R

```diff
@@ -365,19 +391,30 @@ mmrm <- function(formula,
     message(paste(fit_msg, collapse = "\n"))
   }
   fit$method <- control$method
-  if (control$method == "Satterthwaite") {
+  fit$cov_method <- control$cov
+  if (identical(fit$method, "Satterthwaite")) {
     covbeta_fun <- h_covbeta_fun(fit)
     fit$jac_list <- h_jac_list(covbeta_fun, fit$theta_est)
-  } else {
+  }
+
+  if (control$cov %in% c("Kenward-Roger", "Kenward-Roger-Linear")) {
     fit$kr_comp <- h_get_kr_comp(fit$tmb_data, fit$theta_est)
     fit$beta_vcov_adj <- h_var_adj(
       v = fit$beta_vcov,
       w = component(fit, "theta_vcov"),
       p = fit$kr_comp$P,
       q = fit$kr_comp$Q,
       r = fit$kr_comp$R,
-      linear = (control$method == "Kenward-Roger-Linear")
+      linear = (control$cov == "Kenward-Roger-Linear")
     )
+  } else if (control$cov %in% c("Empirical", "Empirical-Jackknife")) {
+    fit$beta_vcov_adj <- h_get_empirical(
+      fit$tmb_data, fit$theta_est, fit$beta_est, fit$beta_vcov, control$cov == "Empirical-Jackknife"
+    )
+    dimnames(fit$beta_vcov_adj) <- dimnames(fit$beta_vcov)
+  } else if (identical(control$cov, "Asymptotic")) {
+  } else {
+    stop("Unrecognized covariance method")
```

Comment:
also the wording in error messages would then be good to adjust to the new name of the argument